Python: Fixed an issue where the schema property was missing when using the OpenAPI plugin with the get method.

ymuichiro opened the pull request:
Fixed an issue where the schema property was missing when using the OpenAPI plugin with the get method.

### Motivation and Context

This Pull Request is based on Issue https://github.com/microsoft/semantic-kernel/issues/8423. When loading an openapi plugin that uses the get method, properties other than the "type" key in parameters.schema is missing.

### Description 

The parameters.schema includes properties other than the "type" key, such as "description", which is used for the llm to make judgments during function calling. Therefore, I modified it to include this information.

[openapi schema](https://github.com/OAI/OpenAPI-Specification/blob/4e9d2b3ec859beef08309c414f895c73a793b958/schemas/v3.0/schema.yaml#L658)

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

moonbox3 commented:
Thanks for doing this, @ymuichiro. Are there any unit tests that need to be added based on the new code?

markwallace-microsoft commented:
<!-- Pytest Coverage Comment: python-tests-coverage | python-test-coverage -->
<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/README.md"><img alt="Python Test Coverage" src="https://img.shields.io/badge/Python Test Coverage-98%25-brightgreen.svg" /></a><details><summary>Python Test Coverage Report </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/kernel.py">kernel.py</a></td><td>199</td><td>47</td><td>76%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/kernel.py#L148">148</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/kernel.py#L159">159</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/kernel.py#L163">163</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/kernel.py#L313-L316">313&ndash;316</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/kernel.py#L423">423</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/kernel.py#L437-L480">437&ndash;480</a></td></tr><tr><td colspan="5"><b>semantic_kernel/agents/channels</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/channels/open_ai_assistant_channel.py">open_ai_assistant_channel.py</a></td><td>49</td><td>1</td><td>98%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/channels/open_ai_assistant_channel.py#L42">42</a></td></tr><tr><td colspan="5"><b>semantic_kernel/agents/group_chat</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/group_chat/agent_chat.py">agent_chat.py</a></td><td>116</td><td>1</td><td>99%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/group_chat/agent_chat.py#L91">91</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/group_chat/agent_group_chat.py">agent_group_chat.py</a></td><td>67</td><td>2</td><td>97%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/group_chat/agent_group_chat.py#L113">113</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/group_chat/agent_group_chat.py#L135">135</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/group_chat/broadcast_queue.py">broadcast_queue.py</a></td><td>72</td><td>1</td><td>99%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/group_chat/broadcast_queue.py#L35">35</a></td></tr><tr><td colspan="5"><b>semantic_kernel/agents/open_ai</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/open_ai/assistant_content_generation.py">assistant_content_generation.py</a></td><td>68</td><td>2</td><td>97%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/open_ai/assistant_content_generation.py#L81-L82">81&ndash;82</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/open_ai/open_ai_assistant_base.py">open_ai_assistant_base.py</a></td><td>361</td><td>3</td><td>99%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/open_ai/open_ai_assistant_base.py#L243">243</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/agents/open_ai/open_ai_assistant_base.py#L321-L322">321&ndash;322</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/azure_ai_inference/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py">azure_ai_inference_chat_completion.py</a></td><td>153</td><td>4</td><td>97%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py#L177">177</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py#L327-L328">327&ndash;328</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py#L444">444</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/embeddings</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/embeddings/embedding_generator_base.py">embedding_generator_base.py</a></td><td>8</td><td>1</td><td>88%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/embeddings/embedding_generator_base.py#L50">50</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/google</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/shared_utils.py">shared_utils.py</a></td><td>39</td><td>2</td><td>95%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/shared_utils.py#L97">97</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/shared_utils.py#L119">119</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/google/google_ai/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/google_ai/services/google_ai_chat_completion.py">google_ai_chat_completion.py</a></td><td>152</td><td>4</td><td>97%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/google_ai/services/google_ai_chat_completion.py#L155">155</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/google_ai/services/google_ai_chat_completion.py#L267">267</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/google_ai/services/google_ai_chat_completion.py#L301-L302">301&ndash;302</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/google_ai/services/utils.py">utils.py</a></td><td>63</td><td>3</td><td>95%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/google_ai/services/utils.py#L139">139</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/google_ai/services/utils.py#L157-L162">157&ndash;162</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/google/vertex_ai/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/vertex_ai/services/utils.py">utils.py</a></td><td>65</td><td>3</td><td>95%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/vertex_ai/services/utils.py#L141">141</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/vertex_ai/services/utils.py#L159-L164">159&ndash;164</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/vertex_ai/services/vertex_ai_chat_completion.py">vertex_ai_chat_completion.py</a></td><td>152</td><td>4</td><td>97%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/vertex_ai/services/vertex_ai_chat_completion.py#L149">149</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/vertex_ai/services/vertex_ai_chat_completion.py#L260">260</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/google/vertex_ai/services/vertex_ai_chat_completion.py#L294-L295">294&ndash;295</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/hugging_face/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py">hf_text_completion.py</a></td><td>54</td><td>3</td><td>94%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py#L97">97</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py#L105">105</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py#L136">136</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_embedding.py">hf_text_embedding.py</a></td><td>32</td><td>5</td><td>84%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_embedding.py#L79-L83">79&ndash;83</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/memory/azure_ai_search</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py">azure_ai_search_collection.py</a></td><td>87</td><td>2</td><td>98%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py#L150">150</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py#L152">152</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/memory/redis</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/memory/redis/redis_collection.py">redis_collection.py</a></td><td>160</td><td>2</td><td>99%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/memory/redis/redis_collection.py#L146">146</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/memory/redis/redis_collection.py#L316">316</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/memory/redis/utils.py">utils.py</a></td><td>45</td><td>11</td><td>76%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/memory/redis/utils.py#L145-L146">145&ndash;146</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/memory/redis/utils.py#L164">164</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/memory/redis/utils.py#L166">166</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/memory/redis/utils.py#L173-L188">173&ndash;188</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/openapi_plugin</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/openapi_plugin/openapi_manager.py">openapi_manager.py</a></td><td>58</td><td>2</td><td>97%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/openapi_plugin/openapi_manager.py#L110-L111">110&ndash;111</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/openapi_plugin/openapi_parser.py">openapi_parser.py</a></td><td>88</td><td>2</td><td>98%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/openapi_plugin/openapi_parser.py#L71">71</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/openapi_plugin/openapi_parser.py#L128">128</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/openapi_plugin/openapi_runner.py">openapi_runner.py</a></td><td>105</td><td>2</td><td>98%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/openapi_plugin/openapi_runner.py#L181-L182">181&ndash;182</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/openapi_plugin/models</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/openapi_plugin/models/rest_api_operation.py">rest_api_operation.py</a></td><td>129</td><td>1</td><td>99%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/connectors/openapi_plugin/models/rest_api_operation.py#L242">242</a></td></tr><tr><td colspan="5"><b>semantic_kernel/contents</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/contents/function_call_content.py">function_call_content.py</a></td><td>97</td><td>1</td><td>99%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/contents/function_call_content.py#L201">201</a></td></tr><tr><td colspan="5"><b>semantic_kernel/data</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/data/vector_store_record_collection.py">vector_store_record_collection.py</a></td><td>249</td><td>19</td><td>92%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/data/vector_store_record_collection.py#L410">410</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/data/vector_store_record_collection.py#L470-L474">470&ndash;474</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/data/vector_store_record_collection.py#L482-L486">482&ndash;486</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/data/vector_store_record_collection.py#L526-L529">526&ndash;529</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/data/vector_store_record_collection.py#L536-L539">536&ndash;539</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/data/vector_store_record_utils.py">vector_store_record_utils.py</a></td><td>26</td><td>2</td><td>92%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/data/vector_store_record_utils.py#L50">50</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/data/vector_store_record_utils.py#L52">52</a></td></tr><tr><td colspan="5"><b>semantic_kernel/functions</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_decorator.py">kernel_function_decorator.py</a></td><td>98</td><td>1</td><td>99%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_decorator.py#L102">102</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_from_method.py">kernel_function_from_method.py</a></td><td>96</td><td>1</td><td>99%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_from_method.py#L153">153</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_from_prompt.py">kernel_function_from_prompt.py</a></td><td>154</td><td>7</td><td>95%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_from_prompt.py#L165-L166">165&ndash;166</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_from_prompt.py#L180">180</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_from_prompt.py#L201">201</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_from_prompt.py#L219">219</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_from_prompt.py#L239">239</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_from_prompt.py#L322">322</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_log_messages.py">kernel_function_log_messages.py</a></td><td>36</td><td>6</td><td>83%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_function_log_messages.py#L37-L43">37&ndash;43</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_plugin.py">kernel_plugin.py</a></td><td>187</td><td>2</td><td>99%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_plugin.py#L472">472</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/functions/kernel_plugin.py#L475">475</a></td></tr><tr><td colspan="5"><b>semantic_kernel/planners</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py">plan.py</a></td><td>234</td><td>45</td><td>81%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L54">54</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L163-L165">163&ndash;165</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L197">197</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L214-L227">214&ndash;227</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L264">264</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L269">269</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L277-L278">277&ndash;278</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L288-L291">288&ndash;291</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L308">308</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L313">313</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L329">329</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L332-L337">332&ndash;337</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L355">355</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L360">360</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L363">363</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L365">365</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L372">372</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L386-L388">386&ndash;388</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/plan.py#L393-L397">393&ndash;397</a></td></tr><tr><td colspan="5"><b>semantic_kernel/planners/function_calling_stepwise_planner</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py">function_calling_stepwise_planner.py</a></td><td>116</td><td>4</td><td>97%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py#L145">145</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py#L189-L190">189&ndash;190</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py#L198">198</a></td></tr><tr><td colspan="5"><b>semantic_kernel/planners/sequential_planner</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner.py">sequential_planner.py</a></td><td>64</td><td>6</td><td>91%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner.py#L71">71</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner.py#L75">75</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner.py#L109">109</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner.py#L125">125</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner.py#L134-L135">134&ndash;135</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner_extensions.py">sequential_planner_extensions.py</a></td><td>50</td><td>9</td><td>82%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner_extensions.py#L31-L32">31&ndash;32</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner_extensions.py#L56">56</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner_extensions.py#L110-L124">110&ndash;124</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner_parser.py">sequential_planner_parser.py</a></td><td>77</td><td>12</td><td>84%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner_parser.py#L66-L74">66&ndash;74</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner_parser.py#L93">93</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/planners/sequential_planner/sequential_planner_parser.py#L117-L120">117&ndash;120</a></td></tr><tr><td colspan="5"><b>semantic_kernel/schema</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/schema/kernel_json_schema_builder.py">kernel_json_schema_builder.py</a></td><td>119</td><td>5</td><td>96%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/schema/kernel_json_schema_builder.py#L50">50</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/schema/kernel_json_schema_builder.py#L84">84</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/schema/kernel_json_schema_builder.py#L206">206</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/schema/kernel_json_schema_builder.py#L210-L211">210&ndash;211</a></td></tr><tr><td colspan="5"><b>semantic_kernel/template_engine/blocks</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/template_engine/blocks/code_block.py">code_block.py</a></td><td>77</td><td>1</td><td>99%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/template_engine/blocks/code_block.py#L119">119</a></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/template_engine/blocks/named_arg_block.py">named_arg_block.py</a></td><td>43</td><td>1</td><td>98%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/template_engine/blocks/named_arg_block.py#L98">98</a></td></tr><tr><td colspan="5"><b>semantic_kernel/utils/telemetry</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/utils/telemetry/user_agent.py">user_agent.py</a></td><td>16</td><td>2</td><td>88%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/utils/telemetry/user_agent.py#L18-L19">18&ndash;19</a></td></tr><tr><td colspan="5"><b>semantic_kernel/utils/telemetry/model_diagnostics</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/utils/telemetry/model_diagnostics/decorators.py">decorators.py</a></td><td>105</td><td>2</td><td>98%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/utils/telemetry/model_diagnostics/decorators.py#L206">206</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/699bc0e452dd0e9cb8fa21832ffc6eaced556f90/python/semantic_kernel/utils/telemetry/model_diagnostics/decorators.py#L210">210</a></td></tr><tr><td><b>TOTAL</b></td><td><b>10829</b></td><td><b>234</b></td><td><b>98%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 2387 | 1 :zzz: | 0 :x: | 0 :fire: | 53.968s :stopwatch: |

ymuichiro commented:
@moonbox3 I added a test pattern now. Added a unit test to verify that openapi.yaml's query and request body are parsed as expected.

https://github.com/microsoft/semantic-kernel/pull/8502/commits/6f10c3a58804d124cdca0d627afc005dfc173eef

moonbox3 reviewed (approved):
Looks great, thank you for doing this and for adding the tests!
